src/lhttpc_lb.erl

```diff
@@ -39,13 +43,16 @@ checkout(Host, Port, Ssl, MaxConn, ConnTimeout) ->
     Lb = find_lb({Host,Port,Ssl}, {MaxConn, ConnTimeout}),
     gen_server:call(Lb, {checkout, self()}, infinity).
 
-%% Called when the socket has died and we're done
--spec checkin(host(), port_number(), SSL::boolean()) -> ok.
-checkin(Host, Port, Ssl) ->
-    case find_lb({Host,Port,Ssl}) of
-        {error, undefined} -> ok; % LB is dead. Pretend it's fine -- we don't care
-        {ok, Pid} -> gen_server:cast(Pid, {checkin, self()})
-    end.
+%% Returns the LB state
```

Comment:
I typically use the fold to avoid copying data from ets.
I concur with your analysis though; I'll rewrite.